src/v/rpc/transport.h

```diff
@@ -159,13 +163,32 @@ template<typename Input, typename Output>
 inline ss::future<result<client_context<Output>>>
 transport::send_typed(Input r, uint32_t method_id, rpc::client_opts opts) {
     using ret_t = result<client_context<Output>>;
+    return send_typed_versioned<Input, Output>(
+             std::move(r), method_id, std::move(opts), transport_version::v0)
+      .then([](result<result_context<Output>> r) {
+          if (!r) {
+              return ss::make_ready_future<ret_t>(r.error());
+          }
+          return ss::make_ready_future<ret_t>(std::move(r.value().ctx));
+      });
```

Comment:
General code style question: is there guidance on when to write functions as coroutines?

Reply:
the general guidance is to write new code as coroutines, and to rewrite old code as coroutines if you feel like it, or if you're going to need to rewrite most of it anyway. but the downside to rewriting a lot of old code as coroutines is the review overhead when a lot of code is changing.